lib/bitpay/client.rb

```diff
@@ -69,12 +69,77 @@ def create_invoice(price:, currency:, facade: 'pos', params:{})
         post(path: "invoices", token: token, params: params)
       end
 
+      ## Gets the privileged merchant-version of the invoice		
+      #   Requires merchant facade token		
+      #		
+      def get_invoice(id:)
+        token = get_token('merchant')
+        get(path: "invoices/#{id}", token: token)
+      end
+
       ## Gets the public version of the invoice
       #
       def get_public_invoice(id:)
         get(path: "invoices/#{id}", public: true)
       end
       
+      
+      ## Refund paid BitPay invoice
+      #
+      #   If invoice["data"]["flags"]["refundable"] == true the a refund address was 
+      #   provided with the payment and the refund_address parameter is an optional override
+      #  
+      #   Amount and Currency are required fields for fully paid invoices but optional
+      #   for under or overpaid invoices which will otherwise be completely refunded
+      #
+      #   Requires merchant facade token
+      #
+      #  @example
+      #    client.refund_invoice(id: 'JB49z2MsDH7FunczeyDS8j', params: {amount: 10, currency: 'USD', bitcoinAddress: '1Jtcygf8W3cEmtGgepggtjCxtmFFjrZwRV'})
+      #
+      def refund_invoice(id:, params:{})
+        invoice = get_invoice(id: id)
```

Comment:
BitPay::BitPayError: 500: Must include amount and currency for refunds of fully paid invoices